### ISSUE TITLE
Update the lwip submodule

### ISSUE THREAD

**Issue by JayFoxRox** [XboxDev/nxdk#130]
Our lwip is from early 2016. There have been some changes which also requires updating the arch specific code.

There's probably various security, stability and performance upgrades we'd benefit from.

**Comment by JayFoxRox** [comment]
Shallow clones of the lwip submodule are problematic. This also affects GitHub Actions CI (which, at the time of writing, is proposed to become our CI). This is not working:

```
- uses: actions/checkout@v1
  with:
    fetch-depth: 1
    submodules: recursive
```

The issue can be recreated locally (although it might not be the exact action/checkout behaviour):

```
$ git init
Initialized empty Git repository in /tmp/lwip/.git/
$ git remote add o https://git.savannah.gnu.org/git/lwip.git
$ git fetch o b9d0d8094638e042f92e15c24015a172cacc65c4
error: Server does not allow request for unadvertised object b9d0d8094638e042f92e15c24015a172cacc65c4
```

The message shows that git.savannah.gnu does not allow fetching of any unadvertised object, so we'd have to use [an existing ref](http://git.savannah.nongnu.org/cgit/lwip.git/refs) instead.
However, we currently use some random commit from early 2016.

Fetching an advertised object (ref) works:
```
$ git fetch o STABLE-2_1_2_RELEASE --depth=1
remote: Counting objects: 438, done.
remote: Compressing objects: 100% (390/390), done.
remote: Total 438 (delta 66), reused 186 (delta 27)
Receiving objects: 100% (438/438), 1.29 MiB | 1.76 MiB/s, done.
Resolving deltas: 100% (66/66), done.
From https://git.savannah.gnu.org/git/lwip
 * tag               STABLE-2_1_2_RELEASE -> FETCH_HEAD
```

The closest tag to our current revision would be http://git.savannah.nongnu.org/cgit/lwip.git/tag/?h=STABLE-2_0_0_RC1
We should at least switch to that revision, to improve the ability to clone.

I also hope that someone knows a better solution, just in case we still want to diverge from lwip tags in the future.
For example, there's also https://git-scm.com/docs/gitmodules#Documentation/gitmodules.txt-submoduleltnamegtshallow to force shallow clones of a submodule (`true`); unfortunately it can not be used to force a deep clone (`false` or similar are unspecified) which could work around this issue.

**Comment by mborgerson** [comment]
Closed by #244
